InfiniTAM/Engine/RosImageSourceEngine.cpp

```diff
@@ -80,41 +86,47 @@ void RosImageSourceEngine::rgbCallback(
 
 void RosImageSourceEngine::depthCallback(
     const sensor_msgs::Image::ConstPtr& msg) {
+  ROS_INFO_ONCE("Got depth raw image.");
   if (!depth_ready_ && data_available_) {
     std::lock_guard<std::mutex> guard(depth_mutex_);
     depth_ready_ = true;
-    // BUG: When getting the stamp from the message tf sais that it cannot
-    // extrapolate to the future. This does not happen all the time but only
-    // with a portion of the messages. tf streaming at 245Hz and camera at 30Hz.
-    // depth_msg_time_stamp_ = msg->header.stamp;
-    depth_msg_time_stamp_ = ros::Time(0);
+    depth_msg_time_stamp_ = msg->header.stamp;
     main_engine_->setImageTimeStamp(depth_msg_time_stamp_.toSec());
-    cv_depth_image_ =
-        cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::TYPE_16UC1);
+
+    // When streaming raw images from Gazebo.
+    if (is_raw_image_TYPE_32FC1_) {
+      cv_depth_image_ =
+          cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::TYPE_32FC1);
+      (cv_depth_image_->image)
+          .convertTo(cv_depth_image_->image, CV_16UC1, 1000);
+    } else {
+      // When doing live streaming from the camera.
+      cv_depth_image_ =
+          cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::TYPE_16UC1);
+    }
```

Comment:
Right now only TYPE_16UC1 and TYPE_32FC1 are supported, is there a better solution?